src/renderer/components/node/Node.tsx

```diff
@@ -147,6 +158,70 @@ const NodeInner = memo(({ data, selected }: NodeProps) => {
     const bgColor = useColorModeValue('gray.400', 'gray.750');
     const shadowColor = useColorModeValue('gray.600', 'gray.900');
 
+    const inputDataValues = useMemo(() => Object.values(inputData), [inputData]);
+
+    const [shouldRun, setShouldRun] = useState(false);
+    useEffect(() => {
+        setShouldRun(
+            !isLocked &&
+                !disabled.isDirectlyDisabled &&
```

Comment:
This didn't fix the disabled bug I reported. Try repeated disabling and enabling a Load Image node.